api/src/pcapi/core/offers/validation.py

```diff
@@ -375,3 +379,20 @@ def check_booking_limit_datetime(
 
     if beginning and booking_limit_datetime and booking_limit_datetime > beginning:
         raise exceptions.BookingLimitDatetimeTooLate()
+
+
+def check_offer_extra_data(offer: Optional[Offer], subcategory_id: str, extra_data: dict) -> dict:
+    api_errors = ApiErrors()
+    subcategory = ALL_SUBCATEGORIES_DICT[subcategory_id]
+    mandatory_fields = list(set(OFFER_EXTRA_DATA_MANDATORY_FIELDS) & set(subcategory.conditional_fields))
+
+    for field in mandatory_fields:
+        if offer and offer.extraData and offer.extraData.get(field):  # type: ignore [union-attr]
+            extra_data[field] = offer.extraData.get(field)  # type: ignore [union-attr]
```

Comment:
Je ne suis pas sûr de bien comprendre la logique, ici. Dans le cas où l'on édite une offre et que ses extraData ont déjà les champs requis remplis, on passe dans la branche "if", donc on conserve les mêmes valeurs et **on ignore les nouvelles valeurs fournies**. Cela veut dire qu'on ne peut pas modifier ces valeurs : est-ce vraiment ce que l'on veut faire ? (Je dirais que non, ça n'est pas ce que l'on veut faire. Mais j'ai peut-être loupé quelque chose ?)

Reply:
Oui les champs genre et type deviennent obligatoires et non modifiables une fois renseigné comme c'est le cas pour la catégorie et la sous-catégorie. Les deux champs deviennent readOnly dans l'édition de l'offre s'ils sont déjà renseignés.
Le reste des champs de extraData reste modifiables.

Reply:
Ok, noté, merci.

(Je m'aperçois que c'était pourtant clairement indiqué dans le titre de la PR. Désolé pour le bruit...)